### PR TITLE
SeaBreeze Logging Support

### PR DESCRIPTION
**Pull request by mathewc**
When running in a linux container, write logs/metrics to stdout. We're logging in the agreed upon CSV format, with a unique identifier for the various event streams.

### CODE REVIEW

File: src/WebJobs.Script/Config/ScriptSettingsManager.cs [hunk 1]
```diff
@@ -44,6 +44,14 @@ public static ScriptSettingsManager Instance
 
         public virtual bool FileSystemIsReadOnly => IsZipDeployment;
 
+        public bool IsLinuxContainer
+        {
+            get
+            {
+                return !IsAzureEnvironment && !string.IsNullOrEmpty(GetSetting(EnvironmentSettingNames.ContainerName));
```
Comment (mathewc) [inline src/WebJobs.Script/Config/ScriptSettingsManager.cs]: Is there a better env var than CONTAINER_NAME to use to determine this?

File: src/WebJobs.Script.WebHost/Diagnostics/LinuxContainerEventGenerator.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
+{
+    internal class LinuxContainerEventGenerator : IEventGenerator
```
Comment (fabiocav) [inline src/WebJobs.Script.WebHost/Diagnostics/LinuxContainerEventGenerator.cs]: Can we just call this `ConsoleEventGenerator` or something along those lines?

File: src/WebJobs.Script/Config/ScriptSettingsManager.cs (hunk 1 above)
Comment (fabiocav) [reply src/WebJobs.Script/Config/ScriptSettingsManager.cs]: Wouldn't this also apply to a non-linux container? Should probably add a platform check here: `RuntimeInformation.IsOSPlatform(OSPlatform.Linux)`

Comment (mathewc) [reply src/WebJobs.Script/Config/ScriptSettingsManager.cs]: The CONTAINER_NAME var is something we're only adding in the SeaBreeze scenario, so should be sufficient. The special handling we require for logs (and other things soon) only come from the SeaBreeze host, so I'm looking to use something specific to that environment.

File: src/WebJobs.Script.WebHost/Diagnostics/LinuxContainerEventGenerator.cs (hunk 2 above)
Comment (mathewc) [reply src/WebJobs.Script.WebHost/Diagnostics/LinuxContainerEventGenerator.cs]: There's another set of changes coming soon where not all of these events will go to console. E.g. the function executed event (currently not implemented) will go to another location. I can explain all of this more in person :)

File: src/WebJobs.Script.WebHost/Diagnostics/LinuxContainerEventGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    {
+        private const string EventTimestampFormat = "MM/dd/yyyy hh:mm:ss.fff tt";
+        private readonly Action<string> _writeEvent;
+
+        public LinuxContainerEventGenerator(Action<string> writeEvent = null)
+        {
+            _writeEvent = writeEvent ?? StdOutWriter;
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/Diagnostics/LinuxContainerEventGenerator.cs]: For testability

File: src/WebJobs.Script.WebHost/Diagnostics/LinuxContainerEventGenerator.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        }
+
+        public static string TraceEventRegex { get; } = $"^{ScriptConstants.LinuxLogEventStreamName} (?<Level>[^,]+),(?<SubscriptionId>[^,]*),(?<AppName>[^,]*),(?<FunctionName>[^,]*),(?<EventName>[^,]*),(?<Source>[^,]*),(?<Details>\".*\"),(?<Summary>\".*\"),(?<HostVersion>[^,]*),(?<EventTimestamp>[^,]+),(?<ExceptionType>[^,]*),(?<ExceptionMessage>\".*\"),(?<FunctionInvocationId>[^,]*),(?<HostInstanceId>[^,]*),(?<ActivityId>[^,]*)$";
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/Diagnostics/LinuxContainerEventGenerator.cs]: Including these here so we can test round tripping of the values. See my tests.

**Pull request merged**